### ISSUE TITLE
Rawhide: Kernel panic for all kola tests in `Openstack` aarch64 runs

### ISSUE THREAD

**Issue by gursewak1997** [coreos/fedora-coreos-tracker#1825]
Every kola test running in `kola-openstack` runs for [rawhide][aarch64] fails with
```
harness.go:106: TIMEOUT[10m0s]: SSH unsuccessful within allotted timeframe for 16f4d302-c567-459d-9549-fb049c3f0293.
harness.go:1740: Found kernel panic (Attempted to kill init! exitcode=0x0000008b) on machine 16f4d302-c567-459d-9549-fb049c3f0293 console
     2024-11-05T18:09:51Z kola: Test timed out. Adding as candidate for rerun success: test_name
```
Last failed run: Openstack [build](https://jenkins-fedora-coreos-pipeline.apps.ocp.fedoraproject.org/blue/organizations/jenkins/kola-openstack/detail/kola-openstack/2821/pipeline)

Example [console.txt](https://github.com/user-attachments/files/17637726/console.txt)
`journal.txt` is empty

**Comment by dustymabe** [comment]
Can you determine when this kernel panic started happening? i.e. what build of FCOS was it introduced in and what was the package diff?

**Comment by gursewak1997** [comment]
Looks like this transition is what "possible" caused this.
`kernel 6.12.0-0.rc1.20241005git27cc6fdf7201.22.fc42` -> `6.12.0-0.rc2.24.fc42`

**Comment by marmijo** [comment]
I saw the same errors when I tested a `[rawhide][aarch64]` image in OpenStack with `kernel-6.12.0-0.rc1.20241005git27cc6fdf7201.22.fc42` pinned.
Example:
```
--- FAIL: podman.base (672.42s)
        harness.go:106: TIMEOUT[10m0s]: SSH unsuccessful within allotted timeframe for 13196846-375b-403e-b51c-f0a9cf868293.
        harness.go:1740: Found kernel panic (Attempted to kill init! exitcode=0x0000000b) on machine 13196846-375b-403e-b51c-f0a9cf868293 console
```
I then tried to launch a VM using that image and saw this in the console log, which is the same as the linked console.txt from above:

```
[    0.000000] Linux version 6.12.0-0.rc1.20241005git27cc6fdf7201.22.fc42.aarch64
...
...
[    6.539248] Run /init as init process
[    6.560245] Kernel panic - not syncing: Attempted to kill init! exitcode=0x0000000b
[    6.567562] CPU: 0 UID: 0 PID: 1 Comm: init Not tainted 6.12.0-0.rc1.20241005git27cc6fdf7201.22.fc42.aarch64 #1
[    6.577147] Hardware name: QEMU KVM Virtual Machine, BIOS 0.0.0 02/06/2015
[    6.583699] Call trace:
[    6.586038]  dump_backtrace+0xdc/0x140
[    6.589632]  show_stack+0x20/0x40
[    6.592812]  dump_stack_lvl+0x60/0x80
[    6.596323]  dump_stack+0x18/0x28
[    6.599508]  panic+0x174/0x390
[    6.602423]  do_exit+0x410/0x418
[    6.605528]  do_group_exit+0x3c/0xa0
[    6.608953]  get_signal+0x81c/0x930
[    6.612312]  do_signal+0x124/0x618
[    6.615574]  do_notify_resume+0xec/0x190
[    6.619353]  el0_da+0x110/0x120
[    6.622375]  el0t_64_sync_handler+0xb4/0x130
[    6.626464]  el0t_64_sync+0x194/0x198
[    6.629984] SMP: stopping secondary CPUs
[    6.633826] Kernel Offset: 0x31b8d9a90000 from 0xffff800080000000
[    6.639644] PHYS_OFFSET: 0x40000000
[    6.642983] CPU features: 0x00,0e000001,40200000,42005203
[    6.648128] Memory Limit: none
[    6.651064] ---[ end Kernel panic - not syncing: Attempted to kill init! exitcode=0x0000000b ]---
```

**Comment by marmijo** [comment]
@gursewak1997 and I spoke a bit and he mentioned that the issue was first introduced in `42.20241009.91.0` because running `42.20241007.91.0` seems to pass in OpenStack without any issues.

The release browser shows this package set change with build `42.20241009.91.0`:

```
Upgraded:

bind-libs 32:9.18.30-1.fc42.aarch64 → 32:9.18.30-2.fc42.aarch64
bind-utils 32:9.18.30-1.fc42.aarch64 → 32:9.18.30-2.fc42.aarch64
cifs-utils 7.0-7.fc41.aarch64 → 7.1-2.fc42.aarch64
ethtool 2:6.10-1.fc41.aarch64 → 2:6.11-1.fc42.aarch64
fedora-release-common 42-0.3.noarch → 42-0.4.noarch
fedora-release-coreos 42-0.3.noarch → 42-0.4.noarch
fedora-release-identity-coreos 42-0.3.noarch → 42-0.4.noarch
git-core 2.46.2-1.fc42.aarch64 → 2.47.0-1.fc42.aarch64
glibc 2.40.9000-1.fc42.aarch64 → 2.40.9000-6.fc42.aarch64
glibc-common 2.40.9000-1.fc42.aarch64 → 2.40.9000-6.fc42.aarch64
glibc-gconv-extra 2.40.9000-1.fc42.aarch64 → 2.40.9000-6.fc42.aarch64
glibc-minimal-langpack 2.40.9000-1.fc42.aarch64 → 2.40.9000-6.fc42.aarch64
google-compute-engine-guest-configs-udev 20240925.00-2.fc42.noarch → 20241004.00-1.fc42.noarch
kernel 6.12.0-0.rc1.20241005git27cc6fdf7201.22.fc42.aarch64 → 6.12.0-0.rc2.24.fc42.aarch64
kernel-core 6.12.0-0.rc1.20241005git27cc6fdf7201.22.fc42.aarch64 → 6.12.0-0.rc2.24.fc42.aarch64
kernel-modules 6.12.0-0.rc1.20241005git27cc6fdf7201.22.fc42.aarch64 → 6.12.0-0.rc2.24.fc42.aarch64
kernel-modules-core 6.12.0-0.rc1.20241005git27cc6fdf7201.22.fc42.aarch64 → 6.12.0-0.rc2.24.fc42.aarch64
lld18-libs 18.1.7-4.fc42.aarch64 → 18.1.8-6.fc42.aarch64
llvm18-libs 18.1.7-5.fc42.aarch64 → 18.1.8-4.fc42.aarch64
p11-kit 0.25.5-3.fc41.aarch64 → 0.25.5-4.fc42.aarch64
p11-kit-trust 0.25.5-3.fc41.aarch64 → 0.25.5-4.fc42.aarch64
podman 5:5.2.3-2.fc42.aarch64 → 5:5.2.4-1.fc42.aarch64
rpm 4.19.94-1.fc42.aarch64 → 4.20.0-1.fc42.aarch64
rpm-libs 4.19.94-1.fc42.aarch64 → 4.20.0-1.fc42.aarch64
rpm-ostree 2024.8-1.fc42.aarch64 → 2024.8-3.fc42.aarch64
rpm-ostree-libs 2024.8-1.fc42.aarch64 → 2024.8-3.fc42.aarch64
rpm-plugin-selinux 4.19.94-1.fc42.aarch64 → 4.20.0-1.fc42.aarch64
shadow-utils 2:4.16.0-4.fc42.aarch64 → 2:4.16.0-5.fc42.aarch64
shadow-utils-subid 2:4.16.0-4.fc42.aarch64 → 2:4.16.0-5.fc42.aarch64
toolbox 0.0.99.6-4.fc42.aarch64 → 0.0.99.6-5.fc42.aarch64
```
We just ruled out that `kernel` is the culprit, so I'll try to run an instance with some other packages pinned to try to find which one introduced the failure. I'll start with `glibc` and `rpm-ostree`.

**Comment by marmijo** [comment]
I pinned `glibc-2.40.9000-1.fc42` (which required `systemd-256.5-1.fc42` to be pinned as well) and I was able to successfully run the instance and kola tests in openstack.
This is the update that caused the regression:
```
glibc 2.40.9000-1.fc42.aarch64 → 2.40.9000-6.fc42.aarch64
glibc-common 2.40.9000-1.fc42.aarch64 → 2.40.9000-6.fc42.aarch64
glibc-gconv-extra 2.40.9000-1.fc42.aarch64 → 2.40.9000-6.fc42.aarch64
glibc-minimal-langpack 2.40.9000-1.fc42.aarch64 → 2.40.9000-6.fc42.aarch64
```

**Comment by dustymabe** [comment]
very interesting..

**Comment by dustymabe** [comment]
just to make sure it's glibc and not systemd, it would be good to know if pinning systemd alone yields a good system.

**Comment by marmijo** [comment]
> just to make sure it's glibc and not systemd, it would be good to know if pinning systemd alone yields a good system.

I ran the tests after pinning only systemd and the failures are still seen.

I also narrowed it down to `glibc-2.40.9000-3.fc42 -> glibc-2.40.9000-4.fc42` as being the transition that caused the failure. Both of these builds were marked 'obsolete' in bodhi so we never saw them in FCOS.
[glibc-2.40.9000-4.fc42](https://bodhi.fedoraproject.org/updates/FEDORA-2024-7bc2bdecbb) saw several aarch64 optimizations that may have caused an issue with OpenStack.

I'll open a BZ today.
